src/components/molecules/search/Search.jsx

```diff
@@ -40,31 +58,91 @@ const Search = () => {
         updateSelectedOption(value);
     };
 
-    const updateTextInput = (event) => {
+    const onChangeHandler = ( event ) => {
         const { value } = event.target;
-        updateSearchText(value);
+        updateSearchText( value );
+        // Make auto suggest API call if search text length >= 3
+        if ( value.length >= 3 ) {
+            setFetchAutoSuggest(true);
+            return;
+        }
+        setFetchAutoSuggest(false);
+    };
+
+    const onSelectHandler = ( value ) => {
+        updateSearchText( value );
     };
 
     return (
         <div className="dictionary-search" data-testid={testIds.SEARCH_CONTAINER}>
             <div className="radio-selection">
-                <Radio label={i18n.startsWith[language]}
+                <Radio label={i18n.startsWithRadioLabel[language]}
                        id="starts-with"
                        className="inline"
-                       value={startsWithValue}
-                       defaultChecked={selectedOption === startsWithValue}
+                       value={searchMatchType.beginsWith}
+                       defaultChecked={selectedOption === searchMatchType.beginsWith}
                        onChange={toggleRadioSelection} />
-                <Radio label={i18n.contains[language]}
+                <Radio label={i18n.containsRadioLabel[language]}
                        id="contains"
                        className="inline"
-                       value={containsValue}
+                       value={searchMatchType.contains}
+                       defaultChecked={selectedOption === searchMatchType.contains}
                        onChange={toggleRadioSelection} />
             </div>
-            <TextInput id="keywords"
-                       action={updateTextInput}
-                       classes="dictionary-search-input"
-                       label=""
-                       placeHolder={i18n.searchPlaceholderText[language]} />
+
+            <Autocomplete
+                id="keywords"
+                label={ i18n.searchPlaceholderText[language] }
+                labelHidden
+                wrapperClasses="dictionary-search-input"
+                inputClasses="dictionary-search-input"
+                value={ searchText }
+                inputProps={{
+                    placeholder: i18n.searchPlaceholderText[language]
+                }}
+                items={ autoSuggest.payload || [] }
+                getItemValue={ item => item.termName }
+                shouldItemRender={ matchItemToTerm }
+                onChange={ event => onChangeHandler( event ) }
+                onSelect={ ( value, item ) => onSelectHandler( value ) }
+                renderMenu={ children => (
+                    <div
+                        className="ncids-autocomplete__menu --terms"
```

Comment:
1. Why `--terms` here?
2. Could you also add a `role=listbox` here?

Reply:
:-) I added it to the test. Will do the same here. Thanks.

Reply:
wait I don't see `role=listbox` anywhere?

Reply:
Sorry, that was a `combobox` and `listbox` will update.

Reply:
Resolved.